### ISSUE TITLE
Implement a nicer required field indicator

### ISSUE THREAD

**Issue by rooby** [govCMS/GovCMS7#172]
The required field indicator "(mandatory)" is not visually very nice.

Would you consider changing it to something similar to Drupal 8's solution, which is an svg of an asterisk added as background on the label, with the field having the required and aria-required attributes?

**Comment by aleayr** [comment]
Referencing #173 to link these together. 

@jozhao commented in the above issue that the (mandatory) is actually part of the accessible_forms module, is there a config to change this at all without forking the module or submitting a patch upstream?

**Comment by pandaskii** [comment]
@aleayr Seems that we have two cases on this issue after a deeper dig

## Case 1 "accessible_forms" module enabled

"accessible_forms" module dose following on form attributes:
- required attribute to required fields
- **required label to required fields (instead of asterix *) - The required field indicator "(mandatory)" here :)**
- aria-invalid to form fields with errors

It can be changed in custom theme level if "accessible_forms" module implements Drupal theme/template methods correctly.

Copy this file "form-required-marker.tpl.php" from "accessible_forms" module to custom theme templates directory. It is a quite simple template, and you can update the codes with your specific requirements. 

``` php
<?php
/**
 * @file
 * Provides an accessible template for form_required_marker.
 */
?>
<span <?php print drupal_attributes($attributes); ?>>(<?php print t('mandatory'); ?>)</span>
```

## Case 2 "accessible_forms" module NOT enabled which is default in govCMS

Add a custom function in custom theme template.php to overwrite govcms_zen_form_required_marker() - [read this link for API explanation](https://api.drupal.org/api/drupal/includes%21form.inc/function/theme_form_required_marker/7)

**Not having context regarding this function in current govCMS base theme, but this function here only replace "*" with "mandatory". @aleayr should we remove this function from current base theme and give it back to custom theme if clients need to alter it**

For example:

```
/**
 * Implements theme_form_required_marker().
 */
function govcms_customtheme_form_required_marker($variables) {
  // This is also used in the installer, pre-database setup.
  $t_function = get_t();
  $attributes = array(
    'class' => 'form-required',
    'title' => $t_function('This field is required.'),
  );
  return '<span' . drupal_attributes($attributes) . '>(' . $t_function('mandatory') . ')</span>';
}
```

**Comment by rooby** [comment]
Since this has been closed, what was the resolution?

**Comment by fiasco** [comment]
@rooby you should be able to implement a theme specific version of the [form_required_marker](https://api.drupal.org/api/drupal/includes%21form.inc/function/theme_form_required_marker/7) theme function. This will allow you to implement SVGs as Drupal 8 does or other. 

If you want this as apart of govCMS, we'd be happy to review a pull request for it!

**Comment by aleayr** [comment]
Hi @rooby, closed because I believed it had been solved and just not updated. I'll triple check, apologies if I've had my wires crossed.

@jozhao Can we confirm we had a solution to this, or have I jumped the gun?

**Comment by rooby** [comment]
Yep, I can confirm that it is possible to fix it with overrides, however it would be nice to have something like Drupal 8's solution out of the box, given the target market for GovCMS.

Happy for it to be closed if that's the decision, it's just that closing without a comment makes it difficult to see at a glance what the resolution was.

**Comment by pandaskii** [comment]
@aleayr yes, we had a solution

I like the thoughts from @rooby considering a [New govCMS default theme built on Bootstrap](https://github.com/govCMS/govCMS/issues/176)

**Comment by fiasco** [comment]
If someone is willing to produce a PR for this, we'll be happy to accept it.

**Comment by nathan-w** [comment]
Question: Why isn't the accessible forms module enabled on govCMS by default?

**Comment by OPCIT** [comment]
Sorry to chime in late.....

Does anyone know what the module actually does?

We've just enabled the module on a site we are redeveloping and saw no difference in the form field output or code.

**Comment by rooby** [comment]
pandaskii outlines its purpose above.

If you use the govcms_zen theme as your base theme then you can disable the module because it replicates the module's functionality.

**Comment by OPCIT** [comment]
We are indeed using the govcms_zem theme as a base, but I see no visual difference to the webform if I have the module enabled or disabled. Either way I get the red '(mandatory)' output and the code is identical.

With module enabled the code output is:

`<label for="edit-submitted-gender">Gender<span class="form-required" title="This field is required.">(mandatory)</span></label>
<select required="true" id="edit-submitted-gender" name="submitted[gender]" class="form-select required"><option value="" selected="selected">- Select -</option><option value="1">Male</option><option value="2">Female</option></select>`

With module disabled code output is:

`<label for="edit-submitted-gender">Gender<span class="form-required" title="This field is required.">(mandatory)</span></label>
 <select required="true" id="edit-submitted-gender" name="submitted[gender]" class="form-select required"><option value="" selected="selected">- Select -</option><option value="1">Male</option><option value="2">Female</option></select>`

So how do we go about removing this rewrite so that the output displays an asterisk instead?

Do we really need to add custom code to the template to achieve this?

**Comment by OPCIT** [comment]
Also, is this module meant to have any configurable elements?

**Comment by OPCIT** [comment]
OK so checking further, it would appear that the module is indeed replicating the govcms_zen theme functionality with replacing the asterisk with the word mandatory. as rooby states.

Using any other theme displays the asterisk when the module is disabled and '(mandatory)' when enabled.

**Comment by rooby** [comment]
There is no configuration. As mentioned before, if you are using the govcms_zen base theme you don't need the module because the theme does the same things.

If you want the default asterisk in your theme then you can override the theme_form_required_marker() . You could use the markup in the default theme function - https://api.drupal.org/api/drupal/includes%21form.inc/function/theme_form_required_marker/7.x

**Comment by rooby** [comment]
As an aside I opened a related issue for the accessible_forms module to find out why we change it to (mandatory) - https://www.drupal.org/project/accessible_forms/issues/2956532
